packages/typography/src/Typography.tsx

```diff
@@ -117,33 +100,60 @@ Subtitle.displayName = 'Subtitle';
 /**
  * Body
  */
-type BodyProps = HTMLElementProps<'div'> & {
+
+type BodyFontWeight = 'regular' | 'medium';
+type BodyProps = HTMLElementProps<'p'> & {
```

Comment:
There are likely some less-common elements that will have valid attributes that will have invalid props according to TS. (e.g. `<label htmlFor="...">` or `<a href="...">`)

Reply:
Would they be using those tags like `<Body as="label" htmlFor="...">` though? How we define BodyProps shouldn't affect children.

Reply:
The way the props are typed currently, `<Body as="label" htmlFor="...">` would be invalid (since `htmlFor` is not a valid `div` prop). The solution likely involves some tricky TS generics to do it right
(or we can just allow any other prop)

I'd start with something like:
```
type BodyProps<T extends keyof JSX.IntrinsicElements> = HTMLElementProps<T> & {
```

Reply:
I think I have a TS solution if you want to sync at some point